core/src/main/scala/cats/FunctorEmpty.scala

```diff
@@ -0,0 +1,28 @@
+package cats
+
+import simulacrum.typeclass
+
+/**
+ * `FunctorEmpty[F]` allows you to `map` and filter out elements simultaneously.
+ */
+@typeclass
+trait FunctorEmpty[F[_]] extends Serializable {
+  def functor: Functor[F]
+
```

Comment:
Can we define any extra methods if we require implementors to implement an `empty` method?  Or just for the laws? The Haskell precedent doesn't seem to define a method like this and I'm not sure if we should either, need to think about what the actual benefits are 🤔

Reply:
well, I guess if you have an empty, since you can't combine it, you are always stuck with an empty. But still, that's interesting for laws `empty.map(fn) == empty` etc...

I hate to have a hacky way to make an empty, but not provide it (namely give me any instance, and I will filter it).

Reply:
Here's some discussion on it https://github.com/typelevel/cats/issues/1365 and here's the haskell counter part http://hackage.haskell.org/package/witherable-0.2/docs/Data-Witherable.html

Reply:
The original PR also has some discussion on why it wasn't included in the first place https://github.com/typelevel/cats/pull/1225/files#discussion_r71987653

Reply:
More specifically, this comment: https://github.com/typelevel/cats/pull/1225#discussion_r71981348

Reply:
Yeah after thinking about it more a bit, I think I'd vote to keep it as is without the empty method.